### ISSUE TITLE
Tag pages post count is occasionally off

### ISSUE THREAD

**Issue by michael-tharrington** [forem/forem#9295]
**Describe the bug**

When visiting a tag's landing page (e.g. dev.to/t/nameoftag) and looking at the number of posts published in the lefthand column, often this number doesn't match up to the actual number of posts there are.

**To Reproduce**

1. Go to https://dev.to/t/javascrip _(NOTE: I'm using a misspelled tag, so we have a lighter count to look at)_
2. Look to the lefthand column for "1 post published"
3. You should be able to count 3 posts on the page

**Expected behavior**

Number of posts published should match up to how many posts there are live on DEV.

**Screenshots**

Number of published posts doesn't match the count.

![Image 2020-07-13 at 2 28 53 PM](https://user-images.githubusercontent.com/16007075/87340089-9dc2dd00-c515-11ea-98e3-48231f05a1eb.png)

**Additional context**

There are some tags that require approval for a post to show up under the tag. For instance, [#shecoded](https://dev.to/t/shecoded) is like this. So, when dealing with tags that require approval, I'd only count and show the articles that are approved.

**Comment by citizen428** [comment]
To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by citizen428** [comment]
Some context (I don't have time right now to look into it more):

1. This gets rendered by `app/views/articles/tags/_sidebar.html.erb:66`:
    ```ruby
    <%= pluralize @num_published_articles, "Post" %> Published
    ```
2. `@num_published_articles` gets set in `StoriesController#handle_tag_index`. 
3. For tags that don't need approval, this uses the `cached_tagged_count` method in the same file, which caches the value for 2h:
    ```ruby
    Rails.cache.fetch("article-cached-tagged-count-#{@tag}", expires_in: 2.hours) do
      Article.published.cached_tagged_with(@tag).where("score > 2").size
    end
    ```

**Comment by atsmith813** [comment]
This is somewhat intentional that they're slightly off due to the caching. The gem isn't the best with keeping those counts. I think what may have happened is when we updated/fixed tags, our counts became even more off. And this count is more obvious on a lightly used tag here where there aren't enough posts to paginate.

I'm curious what people think about trying this in a production console:
```ruby
ActsAsTaggableOn::Tag.find_each do |tag|
  ActsAsTaggableOn::Tag.reset_counters(tag.id, :taggings)
end
```

And seeing how that looks cc: @maestromac

**Comment by mikeydorje** [comment]
Closing as stale. Known caching issue with acts-as-taggable-on gem. July 2020 discussion identified the cause (2-hour cache, counter maintenance) and suggested periodic counter resets, but no action taken. No activity since 2021.

If tag counts are significantly incorrect, please open a new issue with current examples.
